### ISSUE TITLE
Modded components/items are not queued in the assemblers

### ISSUE THREAD

**Issue by LennyIndustries** [Gorea235/TaledensInvManagerUpdated#19]
For some reason modded components/items are not queued in the assembler(s). I tested with production control component from super modules and shield modules from Energy Shields.

**Comment by K1ll3rM** [comment]
You probably need to add them to the item list in the script itself

**Comment by jsomm** [comment]
Did this ever get resolved? I'm currently having the same issue and am unsure how to add the modded components to the item list in the script.

**Comment by LennyIndustries** [comment]
Yes, in the code is a list with all components and stuff, you need to add it there to the script, recompile and run. It then should show up, it did work for me, if you need any more help, just leave a comment or contact me on discord, Lenny#1239

**Comment by ojacobson** [comment]
I took a look at this last night for unrelated reasons. It turns out this is _really hard_, and needs careful consideration. There's a reason Isy's Inventory Manager requires that you manually queue a large stack of modded items before it can figure out how to build them.

In order to queue items in the assembler, you need to know the name of a "blueprint" for that item. The blueprint name doesn't automatically have anything to do with the item name, and there's no API for getting the blueprints given an existing item. You _can_ get the blueprint for an item that's already queued in the assembler, but even then, identifying what item that blueprint will create is challenging. Isy's script appears to correlate blueprints and items by watching inventory counts and ruling out items it already knows about. TIM has no such logic.

The giant table at the head of the script, which begins with `AmmoMagazine/`, contains information about the blueprints known to TIM. In cases where the blueprint name is exactly the same as the item name, you can add entries to this table by adding a line with `/ItemName` to that table. If the name of the blueprint is different, you need to add `/ItemName,,,,BlueprintName`, instead. (The empty values are for default quotas and for replacing the item name with a simpler name.) This workaround should get you moving - adding automatic blueprint detection to TIM would be a major undertaking.
